rotkehlchen/exchanges/manager.py

```diff
@@ -82,58 +88,88 @@ def edit_exchange(
-                cursor,
-                name=name,
-                location=location,
-                new_name=new_name,
-                api_key=api_key,
-                api_secret=api_secret,
-                passphrase=passphrase,
-                kraken_account_type=kraken_account_type,
-                binance_selected_trade_pairs=binance_selected_trade_pairs,
-                ftx_subaccount=ftx_subaccount,
-            )
+        # First validate exchange credentials
+        edited = exchangeobj.edit_exchange_credentials(ExchangeAuthCredentials(
+            api_key=api_key,
+            api_secret=api_secret,
+            passphrase=passphrase,
+            ftx_subaccount=ftx_subaccount,
+        ))
+        if edited is True:
+            try:
+                credentials_are_valid, msg = exchangeobj.validate_api_key()
+            except Exception as e:  # pylint: disable=broad-except
+                msg = str(e)
+                credentials_are_valid = False
+
+            if credentials_are_valid is False:
+                exchangeobj.reset_to_db_credentials()
+                return False, f'New credentials are invalid. {msg}'
+
+        # Then edit extra properties
+        success, msg = exchangeobj.edit_exchange_extras({
+            KRAKEN_ACCOUNT_TYPE_KEY: kraken_account_type,
+            BINANCE_MARKETS_KEY: binance_selected_trade_pairs,
+        })
+        if success is False:
+            exchangeobj.reset_to_db_credentials()
+            return False, f'Failed to edit exchange extras. {msg}'
 
-            # Edit the exchange object
-            success, msg = exchangeobj.edit_exchange(
-                name=new_name,
-                api_key=api_key,
-                api_secret=api_secret,
-                passphrase=passphrase,
-                kraken_account_type=kraken_account_type,
-                binance_selected_trade_pairs=binance_selected_trade_pairs,
-                ftx_subaccount=ftx_subaccount,
-            )
-            if success is False:
-                # by raising we also rollback the DB writes due to user_write()
-                raise InputError(msg)
+        try:
+            with self.database.user_write() as cursor:
+                self.database.edit_exchange(
+                    cursor,
+                    name=name,
+                    location=location,
+                    new_name=new_name,
+                    api_key=api_key,
+                    api_secret=api_secret,
+                    passphrase=passphrase,
+                    kraken_account_type=kraken_account_type,
+                    binance_selected_trade_pairs=binance_selected_trade_pairs,
+                    ftx_subaccount=ftx_subaccount,
+                )
+        except InputError as e:
+            exchangeobj.reset_to_db_credentials()  # DB is already rolled back at this point
+            exchangeobj.reset_to_db_extras()
+            return False, f'Coudnt update exchange properties in the DB. {str(e)}'
```

Comment:
There is a typo but also 

```suggestion
            return False, f"Couldn't update exchange properties in the DB. {str(e)}"
```